### PR TITLE
Fixes for release bugs

### PR DESCRIPTION
**Pull request by ms-mikeb**
## Changes
1. The first issue was the improper use of reusable workflows. The workflow is supposed to identify what it `runs-on` - but I had it using `runs-on` in the job that was calling the workflow, which was making it not function like a workflow.
2. Additional testing revealed there was some script stuff not working as expected. I also resolved those issues.

### CODE REVIEW

**Review by chpence — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -21,4 +22,5 @@ jobs:
     - name: Push Stable Package
       run: |
         $fileName = (Get-ChildItem -Recurse -Filter *.nupkg | Select -Property Name).Name
-        nuget push signed/${{fileName}} -Source https://api.nuget.org/v3/index.json -ApiKey ${{ secrets.api_key }} -Verbosity detailed -NonInteractive
+        nuget push signed/$fileName -Source https://api.nuget.org/v3/index.json -ApiKey ${{ secrets.api_key }} -Verbosity detailed -NonInteractive
```
Comment (amdawood) [inline .github/workflows/publish.yml:25]: I hope you've tested this and 100% sure it's going to work.

**Review by amdawood — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (ms-mikeb) [reply .github/workflows/publish.yml:25]: I mean, I didn't test this exact call, I tested with a `Write-Output $fileName` to ensure the correct variable was output. `${{fileName}}` did not work. It seems the `${{variable}}` syntax is only for pipeline variables interpolation, not script variables.

Comment (amdawood) [reply .github/workflows/publish.yml:25]: Also be mindful of the syntax difference between PowerShell and Bash (jobs running Ubuntu default to Bash script)

Comment (ms-mikeb) [reply .github/workflows/publish.yml:25]: Yup, this is why I overrode the default shell with `shell: pwsh` to allow the use of the powershell cmdlets.

**Pull request merged**